enclave/src/lib.rs

```diff
@@ -425,6 +424,54 @@ pub unsafe extern "C" fn accept_pdex_accounts(
     sgx_status_t::SGX_SUCCESS
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn send_disk_data(encoded_data: *const u8, data_size: usize) -> sgx_status_t {
+    let mut data = slice::from_raw_parts(encoded_data, data_size);
+
+    let decoded: polkadex_sgx_primitives::StorageData =
+        if let Ok(data) = polkadex_sgx_primitives::StorageData::decode(&mut data) {
+            data
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+
+    let balances_data = decoded.balances;
+    let nonce_data = decoded.nonce;
+    let orderbook_data = decoded.orderbook;
+
+    if polkadex_balance_storage::load_balance_storage().is_err()
+        && polkadex_balance_storage::create_in_memory_balance_storage().is_err()
+    {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    if accounts_nonce_storage::load_registry().is_err()
+        && accounts_nonce_storage::create_in_memory_accounts_and_nonce_storage(vec![]).is_err()
+    {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    if polkadex_orderbook_storage::load_orderbook().is_err()
+        && polkadex_orderbook_storage::create_in_memory_orderbook_storage(vec![]).is_err()
+    {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    if polkadex_balance_storage::lock_storage_extend_from_disk(balances_data).is_err() {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    if accounts_nonce_storage::lock_nonce_storage_extend_from_disk(nonce_data).is_err() {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    if polkadex_orderbook_storage::lock_storage_extend_from_disk(orderbook_data).is_err() {
+        return sgx_status_t::SGX_ERROR_UNEXPECTED;
+    }
+
+    sgx_status_t::SGX_SUCCESS
+}
+
```

Comment:
How would it look if we wrap this function in an inner retruning errors with `?` and an outer which will remap all errors to `sgx_status_t::SGX_ERROR_UNEXPECTED` in a `match` ?

Reply:
Those function are all defined in the .edl file and are directly called from the ecall. So a single "outer" function will not work. Or how would you implement that?

Reply:
Something like what i pushed now?

Reply:
Yes!